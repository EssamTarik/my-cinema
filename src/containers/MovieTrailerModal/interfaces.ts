import { IMovie } from "../../store/movieSearch/models";
import { IMovieTrailerFetchState } from "../../store/movieTrailer/models";

export interface IProps {
  movie: IMovie;
  trailerState: IMovieTrailerFetchState;
  getMovieTrailer: (movieId: number) => void;
  onRequestClose: () => void;
  resetMovieTrailer: () => void;
}