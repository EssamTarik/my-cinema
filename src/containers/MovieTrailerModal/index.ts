import { connect } from 'react-redux';
import { getMovieTrailer, resetMovieTrailer } from '../../store/movieTrailer/actions';
import { IStateTree } from '../../store/interfaces';
import MovieTrailerModal from './MovieTrailerModal';

const mapStateToProps = ({ trailer }: IStateTree) => ({ trailerState: trailer })
const mapDispatchToProps = {
  getMovieTrailer,
  resetMovieTrailer
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieTrailerModal)