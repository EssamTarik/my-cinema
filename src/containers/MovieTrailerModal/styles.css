.movie-trailer-modal__not-found {
  background-color: black;
  color: white;
  border-radius: 5px;
  padding: 10px;
  font-size: 25px;
}