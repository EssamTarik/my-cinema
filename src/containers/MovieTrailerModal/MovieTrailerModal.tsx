import React, { useEffect } from "react";
import { FormattedMessage } from "react-intl";
import { Modal, MovieTrailer } from "../../components";
import { IProps } from "./interfaces";
import "./styles.css";

const MovieTrailerModalContainer = ({
  resetMovieTrailer,
  onRequestClose,
  movie: { id: movieId },
  trailerState: { data: trailer, error, isFetched },
  getMovieTrailer,
}: IProps) => {
  useEffect(() => {
    getMovieTrailer(movieId);
  }, [movieId, getMovieTrailer]);

  useEffect(() => {
    return () => {
      resetMovieTrailer();
    };
  }, [resetMovieTrailer]);

  const noVideos = isFetched && !trailer;
  return (
    <Modal onRequestClose={onRequestClose}>
      {(noVideos || error) && (
        <div className="movie-trailer-modal__not-found">
          <FormattedMessage id="movieTrailer.notFound" />
        </div>
      )}
      {trailer && <MovieTrailer trailer={trailer} />}
    </Modal>
  );
};

export default MovieTrailerModalContainer;
