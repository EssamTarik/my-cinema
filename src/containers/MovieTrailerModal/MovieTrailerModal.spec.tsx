import React from "react";
import { render, cleanup } from "@testing-library/react";

import MovieTrailerModal from "./MovieTrailerModal";

jest.mock("../../components", () => ({
  Modal: ({
    children,
  }: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  >) => <div>{children}</div>,
  MovieTrailer: () => <div />,
}));
jest.mock("react-intl", () => ({ FormattedMessage: () => <div /> }));
const mockMovie = jest.requireMock("movie");
const mockTrailerResult = jest.requireMock("trailerResult");

describe("MovieSearchResultsContainer", () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <MovieTrailerModal
        movie={mockMovie}
        trailerState={{
          isFetched: true,
          isFetching: false,
          data: mockTrailerResult,
        }}
        getMovieTrailer={jest.fn()}
        onRequestClose={jest.fn()}
        resetMovieTrailer={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls getMovieTrailer on mount", () => {
    const mockGetMovieTrailer = jest.fn();
    render(
      <MovieTrailerModal
        movie={mockMovie}
        trailerState={{
          isFetched: true,
          isFetching: false,
          data: mockTrailerResult,
        }}
        getMovieTrailer={mockGetMovieTrailer}
        onRequestClose={jest.fn()}
        resetMovieTrailer={jest.fn()}
      />
    );
    expect(mockGetMovieTrailer).toHaveBeenCalledWith(mockMovie.id);
  });

  it("returns not found on empty or error", () => {
    const { container } = render(
      <MovieTrailerModal
        movie={mockMovie}
        trailerState={{
          isFetched: true,
          isFetching: false,
          data: undefined,
          error: "error",
        }}
        getMovieTrailer={jest.fn()}
        onRequestClose={jest.fn()}
        resetMovieTrailer={jest.fn()}
      />
    );
    const notFoundDiv = container.querySelector(
      ".movie-trailer-modal__not-found"
    );
    expect(notFoundDiv).not.toEqual(null);
  });
});
