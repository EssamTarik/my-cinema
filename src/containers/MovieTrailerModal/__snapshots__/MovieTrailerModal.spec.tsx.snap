// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchResultsContainer renders 1`] = `
<DocumentFragment>
  <div>
    <div />
  </div>
</DocumentFragment>
`;
